test/common/network/listen_socket_impl_test.cc

```diff
@@ -67,8 +67,42 @@ TEST_P(ListenSocketImplTest, BindSpecificPort) {
   EXPECT_EQ(addr->asString(), socket3.localAddress()->asString());
 }
 
+TEST_P(ListenSocketImplTest, BindSpecificUDPPort) {
+  // Pick a free port.
+  auto addr_fd = Network::Test::bindFreeLoopbackPort(version_, Address::SocketType::Datagram);
+  auto addr = addr_fd.first;
+  EXPECT_LE(0, addr_fd.second);
+
+  // Confirm that we got a reasonable address and port.
+  ASSERT_EQ(Address::Type::Ip, addr->type());
+  ASSERT_EQ(version_, addr->ip()->version());
+  ASSERT_LT(0U, addr->ip()->port());
+
+  // Release the socket and re-bind it.
+  // WARNING: This test has a small but real risk of flaky behavior if another thread or process
+  // should bind to our assigned port during the interval between closing the fd and re-binding.
+  // TODO(jamessynge): Consider adding a loop or other such approach to this test so that a
+  // bind failure (in the UdpListenSocket ctor) once isn't considered an error.
```

Comment:
```suggestion
  // bind failure (in the UdpListenSocket ctor) once it isn't considered an error.
```

Reply:
@cmluciano  Thanks for the suggestion. Do you want to rebase and update the change?

Reply:
Does it show in the browser an option for just merging it in? If not, I can resubmit to your fork.